### DISCUSSION TITLE
Hijacking Dataview code block for complex query

### DISCUSSION THREAD

**Discussion by ooker777** [SkepticMystic/breadcrumbs#381] [Ideas]
Why don't we just hijack the Dataview codeblock and adding the Breadcrumbs syntax on top of it? For example, say I have this DV codeblock:
```
```dataview
TABLE
  time-played AS "Time Played",
  length AS "Length",
  rating AS "Rating"
FROM #game
SORT rating DESC
```
then the BC codeblock would be:
```
```breadcrumbs
TABLE
  time-played AS "Time Played",
  length AS "Length",
  rating AS "Rating"
FROM #game
SORT rating DESC
START "blabla" 
DIR up
FIELD field1, field2
```
To achieve the default output display of the current BC codeblock, we can use `LIST` and `STAIR` (opposite of `FLAT`, which is also easily mistaken with DV `FLATTEN`) :
```
```breadcrumbs
LIST
START "blabla" 
DIR up
STAIR true
```

This way we only need to learn one thing, but we can use the power of Dataview.

**Comment by michaelpporter** [comment]
Thanks for the detailed proposal! With where v4 landed, this specific approach — making the `breadcrumbs` codeblock speak Dataview's DQL — isn't something we'll pursue, because the project moved in the opposite direction.

**Why it no longer fits:**

- v4 deliberately **dropped the Dataview hard-dependency** (native inline-field parsing landed in 4.14.0). Building the codeblock on top of DQL would re-introduce exactly that dependency.
- The `breadcrumbs` codeblock is now backed by the **native Rust/WASM graph engine** with its own schema (`fields`, `depth`, `sort`, `type`, `show-attributes`, …) rather than a DQL parser. The `STAIR`/`FLAT`/`LIST` ideas already exist as the `flat` option + `type: tree`.

**What covers the "complex query" need today:**

- **Codeblock options** handle most of it: `sort`, `depth`, `fields`, `field-groups`, `show-attributes`, and `type: tree | mermaid | markmap`. See the [codeblock docs](https://breadcrumbs-docs.michaelpporter.com/views/codeblocks/).
- For full **tabular / DQL output**, keep using a normal `dataview` / `dataviewjs` block and pull Breadcrumbs edges through the [API](https://breadcrumbs-docs.michaelpporter.com/api/) — e.g. `window.BCAPI.get_neighbours(path)` or `window.BCAPI.create_list_index()`. That gives you Dataview's power without coupling Breadcrumbs' core to it.
- The optional **`dataview_note`** builder still lets a Dataview query *define edges* (the input side) if you have Dataview installed.

So the "hijack the Dataview codeblock" design is effectively superseded by the native engine + API split. Closing as outdated — but if you build something neat with the API, **Show and tell** would love to see it. 🍞
